src/kernel-install/kernel-install

```diff
@@ -65,14 +65,19 @@ done
 
 if [[ "${0##*/}" == 'installkernel' ]]; then
     COMMAND='add'
+    # make install doesn't pass any parameter wrt initrd handling
+    INITRD_OPTIONS=()
 else
     COMMAND="$1"
     shift
+    INITRD_OPTIONS=( "${@:3}" )
 fi
 
 KERNEL_VERSION="$1"
 KERNEL_IMAGE="$2"
-INITRD_OPTIONS_START="3"
+# If no initrd option is supplied, fallback to "initrd" which is
+# the name used by dracut when generating it in its kernel-install hook
+[[ ${#INITRD_OPTIONS[@]} == 0 ]] && INITRD_OPTIONS=( initrd )
```

Comment:
I guess this works fine, but looks tricky... I think, 90-loaderentry.install should fallback to the previous behavior when no initrd is passed.

Reply:
Last commit solves that I think

Reply:
But just setting 'initrd' in kernel-install is so tricky. 90-loaderentry.install has the following.
```
for initrd in "${@:${INITRD_OPTIONS_START}}"; do
    if [[ -f "${initrd}" ]]; then
        initrd_basename="$(basename ${initrd})"
```
But this block does not used when dracut is used. And the fallback 'initrd' works fine if the current directory does not have 'initrd'. That sounds dangerous for me...

Reply:
I'm not sure I see what you mean by tricky or by dangerous?  The `90-loaderentry.install` is checking the existence of the file file, which it should do regardless of where the default lives (in case the user specifies an invalid file).  Relying upon that check to make the decision for the user (based not on what the user asked for on the command-line, but instead on the existence of a file) seems like the tricky part to me, and the further up out of the guts of the component, the easier that functionality will be to either remove or section off as a compatibility option only.

Could you offer an example of where such a problem occurs?  For me, if a user had an `initrd` in the right directory by accident (potentially an old one) it would get copied in and installed without intention.  When they come to see why and look in `kernel-install` it looks as though nothing unusual is passed to `90-loaderentry.install`, so then they then have to go investigate the semantics of the `90-loaderentry.install` file to figure out why.  I also suspect that this functionatlity needs documenting as a default in a man page, but then documenting how `90-loaderentry.install` works seems like it would require another section or page (rather than just an addition to the `kernel-install` man page).

As I say, I'm happy with the patch as, I don't want this discussion to block it, I just wanted to be clear as to why we've chosen the option of explicitly putting it in the `90-loaderentry.install` file over the other.  5:)

Reply:
Sorry, you've confused me.  My example was what could happen if `90-loaderentry.install` decided to use initrd without the user asking for it.  My suggestion was to make the parameters passed to `90-loaderentry.install` more explicit and move the assumptions out of the loader file and into the `kernel-install` program, so that both `50-dracut.install` and `90-loaderentry.install` get explicitly told which initrd files to use.

If `50-dracut.install` has a dependency on `90-loaderentry.install` to behave a certain way, then that seems to somewhat defeat the purpose of making them independent scripts.  There'd presumably need to be a defined protocol for altering the parameters by something higher in the chain to something lower in the chain?

Is the creation/maintenance of the initrd supposed to be handled by the user, or by `.install` scripts?  My understanding was that the install scripts were simply to put the appropriate files in the appropriate places.  Have I misunderstood their intended purpose?

Reply:
Oh, I see.  So (dracut)[https://github.com/dracutdevs/dracut/blob/master/50-dracut.install] does create the initrd, and then relies on 90-loaderentry.install to decide to install initrd if it exists.  That's troubling (particularly since it allows for initrd to have two different names), so I guess this will need to be sorted out separately between the two projects...